src/T2DMap.cpp

```diff
@@ -4032,15 +4032,26 @@ void T2DMap::slot_setArea()
         return;
     }
 
-    QMapIterator<int, QString> it(mpMap->mpRoomDB->getAreaNamesMap());
-    while (it.hasNext()) {
-        it.next();
-        int areaID = it.key();
-        if (areaID > 0) {
-            arealist_combobox->addItem(QStringLiteral("%1 (%2)").arg(it.value(), QString::number(areaID)), QVariant(areaID));
-        }
+    QStringList sortedAreaList;
+    sortedAreaList = mpMap->mpRoomDB->getAreaNamesMap().values();
+
+    QCollator sorter;
+    sorter.setNumericMode( true ); // Default is false
+    sorter.setCaseSensitivity( Qt::CaseInsensitive ); // Default is case sensitive
+
+    std::sort( sortedAreaList.begin(), sortedAreaList.end(), [&]( const QString& a, const QString& b ) {
```

Comment:
```suggestion
    std::sort(sortedAreaList.begin(), sortedAreaList.end(), sorter);
```

I think that will do the trick according to https://stackoverflow.com/a/36018397/72944. Test?

Reply:
applied